packages/backend/src/core/DiscoveryWatcher.ts

```diff
@@ -69,19 +81,25 @@ export class DiscoveryWatcher {
     this.logger.info('Update finished', { blockNumber })
   }
 
-  async compareWithCommitted(
+  async findChanges(
     name: string,
-    discovered: AnalyzedData[],
+    discovery: ProjectParameters,
     overrides?: Record<string, DiscoveryContract>,
   ): Promise<DiscoveryDiff[]> {
     const committed = await this.configReader.readDiscovery(name)
-    const parsedDiscovery = parseDiscoveryOutput(discovered)
+    const databaseEntry = await this.repository.findLatest(name)
+
+    const currentContracts = databaseEntry
+      ? databaseEntry.discovery.contracts
+      : committed.contracts
```

Comment:
We can skip reading local files until we know that the repository returned undefined.